src/Rings/mpoly-graded.jl

```diff
@@ -1459,7 +1459,17 @@ function monomial_basis(W::MPolyDecRing, d::FinGenAbGroupElem)
      #Ax = b, Cx >= 0
      C = identity_matrix(ZZ, ngens(W))
      A = reduce(vcat, [x.coeff for x = W.d])
-     k = solve_mixed(transpose(A), transpose(d.coeff), C)
+
+     k = try
+       solve_mixed(transpose(A), transpose(d.coeff), C)
+     catch e
+       if e isa ArgumentError && e.msg == "Polyhedron not bounded"
+         rethrow(ArgumentError("The considered graded component is infinite-dimensional"))
```

Comment:
Maybe this could do
```suggestion
         throw(AbstractAlgebra.InfiniteDimensionError())
```
Then in all the other places one could do `if e isa AbstractAlgebra.InfiniteDimensionError` instead of checking the precise error message.

Reply:
Two things:

- replacing `rethrow` by `throw` gives a much longer error message;
- the error message is now ''Infinite-dimensional vector space" which is less precise than the original one.

Is your suggestion for the sake of better code maintenance? Or?

Reply:
Yes, this was about code maintenance. I don't care too much about `throw` or `rethrow`. I understand the argument about the content of the error message.
I assume we can add the possibility to throw an `InfiniteDimensionError` with a custom error message. But that would happen in AbstractAlgebra and require a release etc. I might do that at some point, but not today.
So feel free to ignore this suggestion.

Reply:
> I assume we can add the possibility to throw an `InfiniteDimensionError` with a custom error message. But that would happen in AbstractAlgebra and require a release etc. I might do that at some point, but not today.
> So feel free to ignore this suggestion.

I leave things as they are for now. Please feel free to implement  a custom error message as suggested.